conda-store-server/conda_store_server/storage.py

```diff
@@ -21,10 +21,17 @@ def fset(
         filename: str,
         artifact_type: schema.BuildArtifactType,
     ):
-        db.add(
-            orm.BuildArtifact(build_id=build_id, key=key, artifact_type=artifact_type)
+        ba = orm.BuildArtifact
+        exists = (
+            db.query(ba)
+            .filter(ba.build_id == build_id)
+            .filter(ba.key == key)
+            .filter(ba.artifact_type == artifact_type)
+            .first()
         )
-        db.commit()
+        if not exists:
+            db.add(ba(build_id=build_id, key=key, artifact_type=artifact_type))
+            db.commit()
```

Comment:
This fixes a pre-existing issue. It's important to fix this now because this PR writes to logs line by line, so a new SQL entry would be created for each line.

Multiple SQL entries add nothing here, it's just an oversight. The format is:

```
1|1|LOGS|logs/6e6d70d9-1713098291-1-test.log
2|1|LOGS|logs/6e6d70d9-1713098291-1-test.log
3|1|LOGS|logs/6e6d70d9-1713098291-1-test.log
4|1|LOGS|logs/6e6d70d9-1713098291-1-test.log
5|1|LOGS|logs/6e6d70d9-1713098291-1-test.log
6|1|LOGS|logs/6e6d70d9-1713098291-1-test.log
7|1|LOGS|logs/6e6d70d9-1713098291-1-test.log
```